htlcswitch/switch.go

```diff
@@ -561,16 +561,81 @@ func (s *Switch) CleanStore(keepPids map[uint64]struct{}) error {
 	return s.attemptStore.CleanStore(keepPids)
 }
 
-// SendHTLC is used by other subsystems which aren't belong to htlc switch
-// package in order to send the htlc update. The attemptID used MUST be unique
-// for this HTLC, and MUST be used only once, otherwise the switch might reject
-// it.
+// SendHTLC is the idempotent, safe entry point for local subsystems to dispatch
+// a payment attempt. It wraps the core, non-idempotent htlc dispatch logic.
+//
+// The errors returned by this method are definitive. If a dispatch error
+// occurs, this method transitions the attempt to a terminal FAILED state
+// before returning the error, guaranteeing it is not in-flight. A return of
+// ErrDuplicateAdd indicates the attempt was already accepted and may be
+// in-flight. This contract allows local subsystems to reliably resume their
+// payment lifecycle after a restart.
 func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID, attemptID uint64,
```

Comment:
I wonder will SendHTLC called from the external ChanRouter ? If not and we used SendOnion for this case, I don't think we need to make SendHTLC idempotent or ?